clang/lib/Sema/SemaInit.cpp

```diff
@@ -864,6 +864,14 @@ InitListChecker::FillInEmptyInitializations(const InitializedEntity &Entity,
       WarnIfMissingField &=
           SemaRef.getLangOpts().CPlusPlus || !hasAnyDesignatedInits(SForm);
 
+      if (OuterILE) {
```

Comment:
Perhaps add a comment here about why this is needed?

Reply:
Okay, sure.

Reply:
Ok, ideally this should be reviewed by someone who knows the code but it LGTM.